### PR TITLE
[DPK] Add functions to accommodate device public key.

### PR DESCRIPTION
**Pull request by weiilug**
1. Add function to store dpk in a separate collection ('dpk') of firestore.
2. Add function to retrieve dpks.
3. Complement the removeCredential function to remove the related dpks as well.
4. Update the main.ts to show the dpk list for each credential

### CODE REVIEW

File: src/libs/credential.ts [hunk 1]
```diff
@@ -38,16 +38,20 @@ export interface StoredCredential {
...
 export async function getCredentials(
   user_id: user_id
 ): Promise<StoredCredential[]> {
-  const results: StoredCredential[] = [];
+  let results: StoredCredential[] = [];
   const refs = await store.collection('credentials')
     .where('user_id', '==', user_id)
     .orderBy('registered', 'desc').get();
   refs.forEach(cred => results.push(<StoredCredential>cred.data()));
+  for (var cred of results) {
```
Comment (agektmr) [inline src/libs/credential.ts:52]: Let's not use `var`

```suggestion
  for (let cred of results) {
```

File: src/libs/credential.ts [hunk 2]
```diff
@@ -38,16 +38,20 @@ export interface StoredCredential {
...
   last_used?: number // last used epoc time,
   clientExtensionResults?: any
+  dpks?: string[] // Device Public Key,
 }
 
 export async function getCredentials(
   user_id: user_id
 ): Promise<StoredCredential[]> {
-  const results: StoredCredential[] = [];
+  let results: StoredCredential[] = [];
```
Comment (agektmr) [inline src/libs/credential.ts:47]: No need to use `let`

```suggestion
  const results: StoredCredential[] = [];
```

**Review by agektmr — CHANGES_REQUESTED** [review]

File: src/libs/webauthn.ts [hunk 3]
```diff
@@ -391,6 +411,25 @@ router.post('/authResponse', authzAPI, async (
 
     storedCred.counter = authenticationInfo.newCounter;
     storedCred.last_used = getNow();
+    
+    if (authenticationInfo && 
+        authenticationInfo.authenticatorExtensionResults &&
+        authenticationInfo.authenticatorExtensionResults.devicePublicKey) {
+      const {dpk} = authenticationInfo.authenticatorExtensionResults.devicePublicKey;
```
[the remark this answers is not in the document]
Comment (agektmr) [reply src/libs/webauthn.ts:418]: This is just my personal convention. Never checked the style guide.

[the remark this answers is not in the document]
Comment (weiilug) [reply src/libs/webauthn.ts]: Okay, it's also reasonable to follow your convention. Just feel free to let me know anything I just missed. Thanks!

File: src/public/scripts/main.ts [hunk 4]
```diff
@@ -329,6 +330,8 @@ const listCredentials = async (): Promise<void> => {
             <dd>${cred.credentialPublicKey}</dd>
             <dt>Credential ID</dt>
             <dd>${cred.credentialID}</dd>
+            <dt>Device Public Key</dt>
```
Comment (agektmr) [inline src/public/scripts/main.ts:333]: Can you make this section conditional?

**Review by agektmr — CHANGES_REQUESTED** [review]

**Pull request merged**